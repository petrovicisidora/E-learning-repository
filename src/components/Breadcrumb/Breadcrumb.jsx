//Breadcrumb

import React from "react";

import {
  StyledBreadcrumbContainer,
  StyledBreadcrumbNavigation,
  StyledBreadcrumbUser,
} from "./Breadcrumb.style";

function Breadcrumb() {
  return (
    <StyledBreadcrumbContainer role="navigation">
      <StyledBreadcrumbNavigation>
        Home Page &gt; e-learning Courses
      </StyledBreadcrumbNavigation>
      <StyledBreadcrumbUser>admin</StyledBreadcrumbUser>
    </StyledBreadcrumbContainer>
  );
}

export default Breadcrumb;
