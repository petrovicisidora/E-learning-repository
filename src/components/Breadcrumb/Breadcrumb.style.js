import styled from "styled-components";

export const StyledBreadcrumbContainer = styled.div`
  width: 100%;
  height: 40px;
  display: flex;
  margin-top: 20px;
`;

export const StyledBreadcrumbNavigation = styled.div`
  width: 70%;
  background-color: white;
  padding-left: 10px;
  line-height: 40px;
`;

export const StyledBreadcrumbUser = styled.div`
  width: 30%;
  background-color: #595256f5;
  color: white;
  line-height: 40px;
  text-align: start;
  padding-left: 20px;
`;
