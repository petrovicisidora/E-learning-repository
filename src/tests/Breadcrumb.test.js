import React from "react";
import { render, screen } from "@testing-library/react";

import Breadcrumb from "../components/Breadcrumb/Breadcrumb";

test("renders breadcrumb correctly", () => {
  render(<Breadcrumb />);

  const breadcrumbContainer = screen.getByRole("navigation");
  const breadcrumbNavigation = screen.getByText(
    "Home Page > e-learning Courses"
  );
  const breadcrumbUser = screen.getByText("admin");

  expect(breadcrumbContainer).toBeInTheDocument();
  expect(breadcrumbNavigation).toBeInTheDocument();
  expect(breadcrumbUser).toBeInTheDocument();
});

test("renders breadcrumb's styles correctly", () => {
  render(<Breadcrumb />);

  const breadcrumbContainer = screen.getByRole("navigation");
  const breadcrumbNavigation = screen.getByText(
    "Home Page > e-learning Courses"
  );
  const breadcrumbUser = screen.getByText("admin");

  expect(breadcrumbContainer).toHaveStyle("margin-top: 20px");
  expect(breadcrumbNavigation).toHaveStyle("background-color: white");
  expect(breadcrumbUser).toHaveStyle("text-align: start");
});

test("matches snapshot", () => {
  const { container } = render(<Breadcrumb />);
  expect(container).toMatchSnapshot();
});
