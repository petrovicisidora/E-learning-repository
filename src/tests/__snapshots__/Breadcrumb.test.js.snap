// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="sc-beySPh lfQFTY"
    role="navigation"
  >
    <div
      class="sc-guDLey jxfJxm"
    >
      Home Page &gt; e-learning Courses
    </div>
    <div
      class="sc-dmyCSP dNelWX"
    >
      admin
    </div>
  </div>
</div>
`;
